github-upload-action-secrets

```diff
@@ -63,12 +82,21 @@ def main():
                         help="Only show which secrets would get uploaded where")
     parser.add_argument('-v', '--verbose', action="store_true", default=False,
                         help="Print verbose information")
-    parser.add_argument("secrets_dir", help="directory with one file per secret")
+
+    megr = parser.add_mutually_exclusive_group()
+    megr.add_argument('--directory',
+                      help="Upload given directory with one file per secret")
+    megr.add_argument('--ssh-keygen', metavar="SECRET_NAME",
+                      help="Generate an SSH key, upload private, public to stdout")
+
     opts = parser.parse_args()
 
     NAME_RE = r'[a-z][-0-9a-z]*'
     REPO_RE = f'{NAME_RE}/{NAME_RE}'
 
+    if not opts.directory and not opts.ssh_keygen:
+        parser.error('Must give exactly one of --directory or --ssh-keygen')
```

Comment:
[add_mutually_exclusive_group()](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group) has a `required` argument which you can set to `True`, and drop this.

Reply:
TIL!  Thanks, in return :)